include/zephyr/logging/log_backend_net.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2023 David Corbeil
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LOG_BACKEND_NET_H_
+#define ZEPHYR_LOG_BACKEND_NET_H_
+
+#include <stdbool.h>
+
+/**
+ * @brief Allows user to set a server IP address at runtime
+ *
+ * @details This function allows the user to set an IP address at runtime before the syslog network
+ *          backend is initialized since the string address gets parsed in log_backend_api::init().
+ *          For this to work, CONFIG_LOG_BACKEND_NET_SERVER_RUNTIME needs to be enabled and is
+ *          mutually exclusive with CONFIG_LOG_BACKEND_NET_SERVER_STATIC.
+ *
+ * @param addr String that contains the IP address.
+ *
+ * @return True if parsing could be done, false otherwise.
+ */
+
+bool log_backend_net_set_addr(const char *addr);
```

Comment:
Question: Should this be guarded by `#if IS_ENABLED(CONFIG_LOG_BACKEND_NET_SERVER_RUNTIME)`?

Reply:
Good catch. Initially I figured that it wasn't necessary because `log_backend_net_set_addr()` is used no matter what but you're totally right that the declaration needs to be guarded.

Reply:
Ah, you're right--maybe just guarded by `CONFIG_LOG_BACKEND_NET`, then.